tests/integration/golang/aim/app/update_app_test.go

```diff
@@ -0,0 +1,101 @@
+//go:build integration
+package run
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type UpdateAppTestSuite struct {
+	suite.Suite
+	client      *helpers.HttpClient
+	appFixtures *fixtures.AppFixtures
+	apps        []*models.App
+}
+
+func TestUpdateAppTestSuite(t *testing.T) {
+	suite.Run(t, new(UpdateAppTestSuite))
+}
+
+func (s *UpdateAppTestSuite) SetupTest() {
+	s.client = helpers.NewAimApiClient(os.Getenv("SERVICE_BASE_URL"))
+
+	appFixtures, err := fixtures.NewAppFixtures(os.Getenv("DATABASE_DSN"))
+	assert.Nil(s.T(), err)
+	s.appFixtures = appFixtures
+
+	s.apps, err = s.appFixtures.CreateTestApps(context.Background(), 10)
+	assert.Nil(s.T(), err)
+}
+
+func (s *UpdateAppTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.appFixtures.UnloadFixtures())
+	}()
+	tests := []struct {
+		name        string
+		requestBody models.App
+	}{
+		{
+			name: "UpdateValidApp",
+			requestBody: models.App{
+				Type: "app-type",
+				State: models.AppState{
+					"app-state-key": "app-state-value",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+
+			var resp models.App
+			err := s.client.DoPutRequest(
+				fmt.Sprintf("/apps/%s", s.apps[0].ID),
+				tt.requestBody,
+				&resp,
+			)
+			assert.Nil(s.T(), err)
+			assert.Equal(s.T(), "app-type", resp.Type)
+		})
+	}
+}
+
+func (s *UpdateAppTestSuite) Test_Error() {
+	defer func() {
+		assert.Nil(s.T(), s.appFixtures.UnloadFixtures())
+	}()
+	tests := []struct {
+		name        string
+		requestBody any
+	}{
+		{
+			name: "UpdateAppWithIncorrectBody",
```

Comment:
can you please change test name from `UpdateAppWithIncorrectBody` -> `UpdateAppWithIncorrectState`, because only `State` incorrect here?